h2/src/main/org/h2/command/Parser.java

```diff
@@ -8029,6 +8029,10 @@ private void parseTableColumnDefinition(CommandWithColumns command, Schema schem
                 parseReferences(ref, schema, tableName);
                 command.addConstraintCommand(ref);
             }
+            String comment = readCommentIf();
```

Comment:
It looks like comment should be parsed **before** `CHECK` and `REFERENCES`.
https://dev.mysql.com/doc/refman/8.0/en/create-table.html
```SQL
column_definition:
    data_type [NOT NULL | NULL] [DEFAULT {literal | (expr)} ]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [COLLATE collation_name]
      [COLUMN_FORMAT {FIXED|DYNAMIC|DEFAULT}]
      [STORAGE {DISK|MEMORY}]
      [reference_definition]
      [check_constraint_definition]
  | data_type
      [COLLATE collation_name]
      [GENERATED ALWAYS] AS (expr)
      [VIRTUAL | STORED] [NOT NULL | NULL]
      [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [reference_definition]
      [check_constraint_definition]
```

Also comment can be already read by `parseColumnForTable()`. In this situation it should not be read second time to forbid definitions like `columnName INT COMMENT 'some comment 1' COMMENT 'some comment 2'`.